### PR TITLE
Big query upload

### PR DESCRIPTION
**Pull request by wanlin31**
Additional data such as memory and cpu usage will be collected for some test run through [OSS benchmark infra](https://github.com/grpc/test-infra), see [prometheus.py](https://github.com/grpc/grpc/blob/3d84874dcd30f263b71c1e84a4ff1b7b34b4798b/tools/run_tests/performance/prometheus.py). There are also startTime and endTime fields added to summary field, see #29034 .

To upload the additional data to the existing BigQuery table, two changes are required:

* BigQuery data schema: need to update the schema to include the new collected data.
* [bq_upload_result.py](https://github.com/grpc/grpc/pull/29197/files#diff-c3c5c61c61d9091f3f42588a31afa5e8c8ae4078cee89c6fa4eb668561e5408c): this file is used to upload the data collected in the driver container. To upload additional data, this file is needed to be updated to parse and upload the additional data to BigQuery table.

### CODE REVIEW

**Comment by ctiller** [comment]
can you say what's changing and why?

**Comment by wanlin31** [comment]
Hi @ctiller We will be collecting additional data(cpu & memory) from prometheus for load tests running by OSS benchmark infrastructure. The change is to upload the additional data to the existing table: modify the existing table schema and the upload script to parse the additional data and upload the the table. I did not not  put the descriptions there since it is still a draft PR, created for easier discussion, the description subjects to changes.  Thanks.

File: tools/run_tests/performance/bq_upload_result.py [hunk 1]
```diff
@@ -29,51 +29,59 @@
 import massage_qps_stats
 
 gcp_utils_dir = os.path.abspath(
-    os.path.join(os.path.dirname(__file__), '../../gcp/utils'))
+    os.path.join(os.path.dirname(__file__), "../../gcp/utils"))
```
Comment (paulosjca) [inline tools/run_tests/performance/bq_upload_result.py:32]: The double quotes everywhere should be changed back to single quotes. Thanks!

Comment (wanlin31) [reply tools/run_tests/performance/bq_upload_result.py:32]: Right, I didn't realize that I have changed them,  fixed. Thank you!

**Review by gnossen — APPROVED** [review]

File: tools/run_tests/performance/scenario_result_schema.json [hunk 2]
```diff
@@ -1953,5 +1953,80 @@
...
+    "mode": "REPEATED",
+    "name": "serversPrometheusData",
+    "type": "RECORD"
+  },
+  {
+    "mode": "NULLABLE",
+    "name": "testDurationSeconds",
+    "type": "FLOAT"
   }
 ]
```
Comment (paulosjca) [inline tools/run_tests/performance/scenario_result_schema.json:2032]: Text files should end in newline.

File: tools/run_tests/performance/bq_upload_result.py [hunk 3]
```diff
@@ -243,6 +248,48 @@ def _populate_node_metadata_from_file(scenario_result, node_info_file):
     scenario_result['nodeMetadata'] = node_metadata
 
 
+def _populate_prometheus_query_results_from_file(scenario_result,
+                                                 prometheus_query_result_file):
+    """Populate the results from prometheus query to Bigquery table """
```
Comment (paulosjca) [inline tools/run_tests/performance/bq_upload_result.py:253]: prometheus -> Prometheus.

**Review by paulosjca — APPROVED** [review]

**Pull request merged**